### PR TITLE
Add data_set_id to Functions.create

### PR DESCRIPTION
**Pull request by haakonvt**
Let me know if this is okey, it would make it much easier to work with for e.g. AIR 😄  @senickel

### CODE REVIEW

File: cognite/experimental/_api/functions.py [hunk 1]
```diff
@@ -49,6 +48,7 @@ def create(
         secrets: Optional[Dict] = None,
         cpu: Number = 0.25,
         memory: Number = 1.0,
+        data_set_id: Optional[int] = None,
```
Comment (pavelzubarev) [inline cognite/experimental/_api/functions.py:51]: The big question for me should we ever expose internal IDs vs. external IDs. I believe external IDs are the way to go and we should not require client code to resolve that explicitly. What do you think, @haakonvt

Comment (haakonvt) [reply cognite/experimental/_api/functions.py:51]: Boy if I agree ;) But have a look at the FilesAPI.upload; it only accepts `data_set_id` 😢

Comment (senickel) [reply cognite/experimental/_api/functions.py:51]: or do both?

**Review by HMEiding — CHANGES_REQUESTED** [review]
I think adding the argument `data_set_id` to `FunctionsAPI.create()` is a bit misleading since it suggests that the `data_set_id` is referring to the _**function**_ and not the file. I would much rather have for example a helper function `FunctionsAPI.upload_file_from_folder()` which could have `data_set_id` as an optional argument and return a `file_id`.

**Comment by haakonvt** [comment]
> I think adding the argument `data_set_id` to `FunctionsAPI.create()` is a bit misleading since it suggests that the `data_set_id` is referring to the _**function**_ and not the file. I would much rather have for example a helper function `FunctionsAPI.upload_file_from_folder()` which could have `data_set_id` as an optional argument and return a `file_id`.

I agree and I like it! I'll see what I can make 🚀

**Comment by haakonvt** [comment]
I'll abandon this one, since branch name is `add-data-set-id-to-functions-create`

**Pull request closed without merging**